src/app/services/auth.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import {AngularFireAuth } from '@angular/fire/auth';
+import { map } from 'rxjs/operators';
+import { auditTrail } from '@angular/fire/database';
+import { auth } from 'firebase';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+
+  constructor(private afsAuth: AngularFireAuth) { }
+
+  loginGoogleUser() {
+    return this.afsAuth.auth.signInWithPopup(new auth.GoogleAuthProvider());
```

Comment:
What `afs` means? Avoid using abrevations

Reply:
You will need to add scopes to let the backend retrieve information about the user... consider the following:

```
doGoogleLogin(): Observable<User> {
    return new Observable<any>(subscriber => {
      const provider = new firebase.auth.GoogleAuthProvider();
      provider.addScope('profile');
      provider.addScope('email');
      this.afAuth.auth
        .signInWithPopup(provider)
        .then(res => {
          if (res.user && res.user['ra']) {
            this.login(res.user['ra']).subscribe(user => {
              subscriber.next(user);
              subscriber.complete();
            });
          }
        });
    });
  }
```

notice that I'm passing `res.user['ra']` to the method `login` ... the attribute `ra` contains the tokenId that the backed require to send the request to firebase... 

the method login looks like:
```
login(tokenId: string): Observable<User> {
  const urlEndpoint = 'http://<backend url>/oauth/token';
  const credentials = btoa('<username>:<password>');
  const httpHeaders = new HttpHeaders({
    'Content-Type': 'application/x-www-form-urlencoded',
    'Authorization': `Basic ${credentials}`
  });
  const params = new URLSearchParams();
  params.set('grant_type', 'firebase');
  params.set('firebase-token-id', tokenId);

  return this.httpClient.post<any>(urlEndpoint, params.toString(), {headers: httpHeaders})
    .pipe(
      tap(response => {
        console.log(response);
        this.accessToken = response.access_token;
        sessionStorage.setItem(this.ACCESS_TOKEN, this.accessToken);
      }),
      map(() => {
        this._user = this.buildUser(this.accessToken);

        return this._user;
      }));
}

private buildUser(accessToken: string): User {
  if (accessToken) {
    accessToken = accessToken.split('.')[1];
    const userInfo: any = JSON.parse(atob(accessToken));
    this._user = new User();

    this._user.username = userInfo.user_name;
    this._user.name = userInfo.name;
    this._user.roles = userInfo.authorities;
    this._user.email = userInfo.email;

    return this._user;
  }
  return null;
}
```